### DISCUSSION TITLE
Feature request: ability to run audio server-side

### DISCUSSION THREAD

**Discussion by falkTX** [maximecb/noisecraft#78] [General]
Hi there, great project!
I have a request, though I fear it might be a bit unusual, so feel free to close if it is completely out of scope for the project.

The idea is to allow audio to run on the server side, instead of client side.
Client would act purely as frontend (though since you already have web audio working, could be left as is too), and communicate with the server to setup the nodes, connections etc.

This would allow for example:
- Able to run it headless
- Allow integration with other modular tools, if audio running on something like JACK or PipeWire
- Allow remote instance setup, running on a beefier PC
- Background processing that does not need a browser tab open/active

If it doesnt fit the project, I understand.
Just thought to ask, in case you see potential in it

**Comment by maximecb** [comment]
Hi falk. I've converted this issue to a discussion because I don't think it's actionable at the moment.

I think there's an argument to be made that audio generation should potentially be able to run offline, outside of a browser. That's useful for testing and potentially other use cases too.

As far as running the audio generation on a completely different machine across a network, I think that might add a lot of complexity and would fall outside of the scope of the project.

**Reply by falkTX** [reply]
Fair enough.
Though the remote setup is not that complex, just something that a program needs to take into account from the start in order to make it work.

Separating audio generation from frontend/view is always a good practice in my view.

**Reply by maximecb** [reply]
The audio generation does happen in a separate thread/process.
